crates/next-core/src/env.rs

```diff
@@ -67,6 +67,10 @@ pub async fn env_for_js(
         "__NEXT_IMAGE_OPTS".to_string() => serde_json::to_string(&image_config).unwrap(),
     };
 
+    for (var, val) in next_config.env.iter() {
+        map.insert(var.clone(), val.clone());
+    }
```

Comment:
This is happening post encoding (via the `EmbeddableProcessEnvVc`), which means a `{ env: "alert('XSS')" }` injects live code and will actually alert the user.